File: react-java/backend/src/main/java/com/clt/hrm/core/administration/company/dtos/SearchCompanyDto.java
package com.clt.hrm.core.administration.company.dtos;

import com.clt.hrm.infra.common.dtos.SearchBaseDto;
import com.fasterxml.jackson.annotation.JsonAlias;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = false)
public class SearchCompanyDto extends SearchBaseDto {
    @JsonAlias("coNm")
    private String searchText;
    private String useFlg;
    private String taxCd;
    private String coTpCd;
    private String coNtn;
}

File: nuxtjs-nestjs/backend/src/apps/administration/companies/dto/search-company.dto.ts
import { IsOptional, IsString } from 'class-validator';
import { SearchBaseDto } from '@infra/common/dto';

export class SearchCompanyDto extends SearchBaseDto {
  @IsOptional()
  @IsString()
  coNm?: string;

  @IsOptional()
  @IsString()
  taxCd?: string;

  @IsOptional()
  @IsString()
  coTpCd?: string;

  @IsOptional()
  @IsString()
  coNtn?: string;
}
